build failure on Ubuntu22.04 from SIGSTKSZ not being an integer literal

```
In file included from /usr/include/signal.h:328,
                 from /home/derek/dr/git/src/core/drlibc/../lib/globals_shared.h:78,
                 from /home/derek/dr/git/src/core/drlibc/drlibc.h:47,
                 from /home/derek/dr/git/src/suite/tests/tools.h:46,
                 from /home/derek/dr/git/src/suite/tests/tools.c:35:
/home/derek/dr/git/src/suite/tests/tools.h:92:25: error: missing binary operator before token "("
   92 | #if defined(AARCH64) && SIGSTKSZ < 16384
      |                         ^~~~~~~~
[1021/1445] Building ASM object suite/tests/CMakeFiles/tools.dir/tools.c_asm.asm.o
```
It must be from this:
```
/usr/include/x86_64-linux-gnu/bits/sigstksz.h:# define SIGSTKSZ sysconf (_SC_SIGSTKSZ)
```

--- a/core/drlibc/drlibc_unix.c
+++ b/core/drlibc/drlibc_unix.c
@@ -509,6 +509,8 @@ const reg_id_t syscall_regparms[MAX_SYSCALL_ARGS] = {
  */
 static size_t page_size = 0;
 
+static size_t auxv_minsigstksz = 0;
+
 /* Return true if size is a multiple of the page size.
  * XXX: This function may be called when DynamoRIO is in a fragile state, or not
  * yet relocated, so keep this self-contained and do not use global variables or
@@ -577,6 +579,18 @@ os_page_size(void)
     return size;
 }
 
+/* With SIGSTKSZ now in sysconf and an auxv var AT_MINSIGSTKSZ we avoid
+ * using the defines and try to lookup the min value in os_page_size_init().
+ */
+size_t
+os_minsigstksz(void)
+{
+#define MINSIGSTKSZ_DEFAULT 2048
+    if (auxv_minsigstksz == 0)
+        return MINSIGSTKSZ_DEFAULT;
+    return auxv_minsigstksz;
+}
+
 void
 os_page_size_init(const char **env, bool env_followed_by_auxv)
 {
@@ -595,12 +609,18 @@ os_page_size_init(const char **env, bool env_followed_by_auxv)
         /* Skip environment. */
         while (*env != 0)
             ++env;
-        /* Look for AT_PAGESZ in the auxiliary vector. */
+        /* Look for AT_PAGESZ and other values in the auxiliary vector. */
         for (auxv = (ELF_AUXV_TYPE *)(env + 1); auxv->a_type != AT_NULL; auxv++) {
             if (auxv->a_type == AT_PAGESZ) {
                 os_set_page_size(auxv->a_un.a_val);
                 break;
             }
+#    ifdef AT_MINSIGSTKSZ
+            else if (auxv->a_type == AT_MINSIGSTKSZ) {
+                auxv_minsigstksz = auxv->a_un.a_val;
+                break;
+            }
+#    endif
         }
     }
 #endif /* LINUX */

--- a/core/drlibc/drlibc_unix.c
+++ b/core/drlibc/drlibc_unix.c
@@ -585,7 +585,11 @@ os_page_size(void)
 size_t
 os_minsigstksz(void)
 {
-#define MINSIGSTKSZ_DEFAULT 2048
+#ifdef AARCH64
+#    define MINSIGSTKSZ_DEFAULT 5120
+#else
+#    define MINSIGSTKSZ_DEFAULT 2048
+#endif
     if (auxv_minsigstksz == 0)
         return MINSIGSTKSZ_DEFAULT;
     return auxv_minsigstksz;

--- a/core/os_shared.h
+++ b/core/os_shared.h
@@ -1,5 +1,5 @@
 /* **********************************************************
- * Copyright (c) 2010-2021 Google, Inc.  All rights reserved.
+ * Copyright (c) 2010-2022 Google, Inc.  All rights reserved.
  * Copyright (c) 2003-2010 VMware, Inc.  All rights reserved.
  * **********************************************************/
 
@@ -452,8 +452,11 @@ get_dynamorio_library_path(void);
 size_t
 os_page_size(void);
 #ifdef UNIX
+/* This also tries to set other auxv values. */
 void
 os_page_size_init(const char **env, bool env_followed_by_auxv);
+size_t
+os_minsigstksz(void);
 #endif
 bool
 get_memory_info(const byte *pc, byte **base_pc, size_t *size, uint *prot);

--- a/core/unix/signal.c
+++ b/core/unix/signal.c
@@ -140,9 +140,10 @@ sig_is_alarm_signal(int sig)
     return (sig == SIGALRM || sig == SIGVTALRM || sig == SIGPROF);
 }
 
-/* we do not use SIGSTKSZ b/c for things like code modification
- * we end up calling many core routines and so want more space
- * (though currently non-debug stack size == SIGSTKSZ (8KB))
+/* We do not use SIGSTKSZ b/c for things like code modification
+ * we end up calling many core routines and so want more space.
+ * Also, SIGSTKSZ is now defined as sysconf(_SC_SIGSTKSZ) and we
+ * cannot invoke libc.
  */
 #define SIGSTACK_SIZE (DYNAMO_OPTION(signal_stack_size))
 
@@ -2299,7 +2300,7 @@ handle_sigaltstack(dcontext_t *dcontext, const stack_t *stack, stack_t *old_stac
             local_stack.ss_sp = NULL;
             local_stack.ss_size = 0;
         } else {
-            if (local_stack.ss_size < MINSIGSTKSZ) {
+            if (local_stack.ss_size < os_minsigstksz()) {
                 *result = ENOMEM;
                 return false;
             }
@@ -3508,7 +3509,8 @@ get_sigstack_frame_ptr(dcontext_t *dcontext, thread_sig_info_t *info, int sig,
                 sp = cur_sp - frame_sz_max - EXECUTE_NATIVE_STACK_USAGE;
             }
             if (USE_APP_SIGSTACK(info, sig)) {
-#define APP_ALTSTACK_USAGE (SIGSTKSZ / 2)
+                /* We avoid SIGSTKSZ as it now calls sysconf in libc. */
+#define APP_ALTSTACK_USAGE (4096)
                 if (sp - APP_ALTSTACK_USAGE < (byte *)info->app_sigstack.ss_sp) {
                     /* There's not enough stack space.  The only solution would be to
                      * re-use DR's frame and limit our own stack usage here, which

--- a/suite/tests/tools.h
+++ b/suite/tests/tools.h
@@ -61,7 +61,6 @@
 #    endif
 #    include <ucontext.h>
 #    include <unistd.h>
-#    include <linux/version.h>
 #    include "../../core/unix/os_public.h"
 #    ifdef X64
 /* XCode 10.1 (probably others too) toolchain wants _STRUCT_MCONTEXT
@@ -90,10 +89,13 @@
 #undef PFX
 #define PFX "0x" PFMT
 
-#if defined(UNIX) && defined(AARCH64) && LINUX_VERSION_CODE < KERNEL_VERSION(4, 3, 0)
+#ifdef LINUX
+#    include <linux/version.h>
+#    if defined(AARCH64) && LINUX_VERSION_CODE < KERNEL_VERSION(4, 3, 0)
 /* SIGSTKSZ was incorrectly defined in Linux releases before 4.3. */
-#    undef SIGSTKSZ
-#    define SIGSTKSZ 16384
+#        undef SIGSTKSZ
+#        define SIGSTKSZ 16384
+#    endif
 #endif
 
 #ifdef __cplusplus

--- a/suite/tests/tools.h
+++ b/suite/tests/tools.h
@@ -61,6 +61,7 @@
 #    endif
 #    include <ucontext.h>
 #    include <unistd.h>
+#    include <linux/version.h>
 #    include "../../core/unix/os_public.h"
 #    ifdef X64
 /* XCode 10.1 (probably others too) toolchain wants _STRUCT_MCONTEXT
@@ -89,7 +90,7 @@
 #undef PFX
 #define PFX "0x" PFMT
 
-#if defined(AARCH64) && SIGSTKSZ < 16384
+#if defined(UNIX) && defined(AARCH64) && LINUX_VERSION_CODE < KERNEL_VERSION(4, 3, 0)
 /* SIGSTKSZ was incorrectly defined in Linux releases before 4.3. */
 #    undef SIGSTKSZ
 #    define SIGSTKSZ 16384

